deployment/tfhub/object-detection-base64.ipynb

```diff
@@ -0,0 +1,627 @@
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Ifi5nJeBX_1G"
+      },
+      "outputs": [],
+      "source": [
+        "urls = urls[:10]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "6puNawPEYBJl"
+      },
+      "outputs": [],
+      "source": [
+        "def process_image(url):\n",
+        "    file_name = os.path.basename(url)\n",
+        "    wget.download(url, file_name)\n",
+        "\n",
+        "    with open(file_name, 'rb') as img:\n",
+        "        img_bytes = base64.b64encode(img.read())\n",
+        "        img_str = img_bytes.decode('utf-8')\n",
+        "        img_str = json.dumps(img_str)\n",
+        "    \n",
+        "    return deployed_model.predict([file_name, img_str])"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "5msNtuwt8AXa"
+      },
+      "outputs": [],
+      "source": [
+        "results = []\n",
+        "start_time = time.time()\n",
+        "\n",
+        "with concurrent.futures.ThreadPoolExecutor() as executor:\n",
+        "    for result in executor.map(process_image, urls):\n",
+        "        results.append(result)\n",
+        "\n",
+        "end_time = time.time()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "jGY1FLwWYBXq"
+      },
+      "outputs": [],
+      "source": [
+        "total_time = end_time - start_time\n",
+        "total_time = time.strftime('%Hh %Mm %Ss', time.gmtime(total_time))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "-k2mErxvYBdZ",
+        "outputId": "e2b67d7f-d951-4efd-d292-0fa6fbd1432a"
+      },
+      "outputs": [],
+      "source": [
+        "print(f\"Processing Time: {total_time} for {len(urls)} images.\")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "sy8guzzgA5df"
+      },
+      "outputs": [],
+      "source": [
+        "results_dfs = []\n",
+        "\n",
+        "for img_result in results:\n",
+        "    cols = list(img_result.keys())[:-1]\n",
+        "    bboxes = list(img_result['bboxes'].keys())\n",
+        "    cols.extend(bboxes)\n",
+        "    data = []\n",
+        "\n",
+        "    for k, v in img_result.items():\n",
+        "        data.extend(list(v.values())) if k == 'bboxes' else data.append(v)\n",
+        "\n",
+        "    results_dfs.append(pd.DataFrame([data], columns = cols))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "XOKZr5qBJqup"
+      },
+      "outputs": [],
+      "source": [
+        "df = pd.concat(results_dfs)"
+      ]
+    },
```

Comment:
Here is a shorter and possibly clearer way to do this:

```python
results_rows = []

for img_result in results:
    img_result = img_result.copy()
    img_result.update(img_result.pop('bboxes'))
    results_rows.append(pd.Series(img_result))

df = pd.concat(results_rows, axis=1).T
```

Reply:
That's really a cleaner code. Done.